src/components/SelectBadge/SelectBadge.tsx

```diff
@@ -18,61 +18,49 @@ const iconMap = {
 
 export type Priority = 'mandatory' | 'important' | 'optional' | 'exclude';
 
-export interface Option {
-    value: string;
+export type PriorityItem<V> = {
+    priority: Priority;
     label: string;
-}
+    value?: V;
+};
 
-export interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+export interface Props<V, O> extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
     /**
      * Children nodes to be rendered within the component,
      * specifically used to display the selected value from parent component.
      */
     children: React.ReactNode;
+    /** Array of availible priority items. */
+    priorityItems?: Array<PriorityItem<V>>;
+    /** Currently selected priority item that indicates the importance of the component. */
+    priorityItem?: PriorityItem<V>;
```

Comment:
I'm wondering why it has name `priorityItem`? but we have `option`, not `optionItem`